src/test/java/org/opensearch/action/admin/indices/mapping/get/ValidateAnomalyDetectorActionHandlerTests.java

```diff
@@ -250,4 +251,63 @@ public void testValidateMoreThanTenMultiEntityDetectorsLimit() throws IOExceptio
         assertTrue(inProgressLatch.await(100, TimeUnit.SECONDS));
         verify(clientSpy, never()).execute(eq(GetMappingsAction.INSTANCE), any(), any());
     }
+
+    // This test also validates that if we get a non timeseries exception or not an invalid query that we will not completely block
+    // detector creation, this is applicable like things when we get timeout not cause of AD configuration errors but because cluster
+    // is momentarily under utilized.
+    public void testValidateMoreThanTenMultiEntityDetectorsLimitDuplicateNameFailure() throws IOException, InterruptedException {
+        SearchResponse mockResponse = mock(SearchResponse.class);
+        int totalHits = 1;
+        when(mockResponse.getHits()).thenReturn(TestHelpers.createSearchHits(totalHits));
+        SearchResponse detectorResponse = mock(SearchResponse.class);
+        when(detectorResponse.getHits()).thenReturn(TestHelpers.createSearchHits(totalHits));
+        SearchResponse userIndexResponse = mock(SearchResponse.class);
+        int userIndexHits = 0;
+        when(userIndexResponse.getHits()).thenReturn(TestHelpers.createSearchHits(userIndexHits));
+        AnomalyDetector singleEntityDetector = TestHelpers.randomAnomalyDetector(TestHelpers.randomUiMetadata(), null, true);
+
+        SearchResponse configInputIndicesResponse = mock(SearchResponse.class);
+        when(configInputIndicesResponse.getHits()).thenReturn(TestHelpers.createSearchHits(2));
+
+        // extend NodeClient since its execute method is final and mockito does not allow to mock final methods
+        // we can also use spy to overstep the final methods
+        NodeClient client = IndexAnomalyDetectorActionHandlerTests
+            .getCustomNodeClient(detectorResponse, userIndexResponse, configInputIndicesResponse, singleEntityDetector, threadPool);
```

Comment:
Can't we just mock the client here?

Reply:
I tried mocking the client here: https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/java/org/opensearch/ad/transport/ValidateAnomalyDetectorTransportActionTests.java#L374
But needed to mock a lot more requests, I changed to use the way we have in this file where we needed to create a customNodeClient where based on the comment it was needed cause we can't mock final methods. I tried also if we just use `client = mock(Client.class);` we can't use spy:
```
Please don't pass mock here. Spy is not allowed on mock.
java.lang.IllegalArgumentException: Please don't pass mock here. Spy is not allowed on mock.
```

Reply:
Yeah that's because we can't spy a mocked object.

I see there's a method already to fetch `getCustomNodeClient` and the only difference here is the featureConfig check. Why not to combine both the method together to reduce code duplication?

Reply:
combined the methods